installer/lib/phx_new/generator.ex

```diff
@@ -83,6 +83,28 @@ defmodule Phx.New.Generator do
     end
   end
 
+  def prod_only_config_inject(path, file, to_inject) do
+    file = Path.join(path, file)
+
+    contents =
+      case File.read(file) do
+        {:ok, bin} -> bin
+        {:error, _} -> """
+          import Config
+
+          if config_env() == :prod do
+        """
+      end
+
+    case split_with_self(contents, "if config_env() == :prod do") do
+      [left, middle, right] ->
+        File.write!(file, [left, middle, ?\n, String.trim_trailing(to_inject), ?\n, right])
+
+      :error ->
+        Mix.raise ~s[Could not find "if config_env() == :prod do" in #{inspect(file)}]
+    end
```

Comment:
similar here, it should always be there!

```suggestion
    [left, middle, right] = split_with_self(contents, "if config_env() == :prod do") do
    File.write!(file, [left, middle, ?\n, String.trim_trailing(to_inject), ?\n, right])
```